specification/cognitiveservices/OpenAI.Inference/models/completions/azure_content_filtering.tsp

```diff
@@ -0,0 +1,165 @@
+  """)
+  sexual?: ContentFilterResult;
+
+  @doc("""
+  Describes language related to physical actions intended to hurt, injure, damage, or 
+  kill someone or something; describes weapons, etc.
+  """)
+  violence?: ContentFilterResult;
+
+  @doc("""
+  Describes language attacks or uses that include pejorative or discriminatory language 
+  with reference to a person or identity group on the basis of certain differentiating 
+  attributes of these groups including but not limited to race, ethnicity, nationality,
+  gender identity and expression, sexual orientation, religion, immigration status, ability
+  status, personal appearance, and body size.
+  """)
+  hate?: ContentFilterResult;
+
+  @doc("""
+  Describes language related to physical actions intended to purposely hurt, injure,
+  or damage one’s body, or kill oneself.
+  """)
+  @projectedName("json", "self_harm")
+  selfHarm?: ContentFilterResult;
+
+  @added(ServiceApiVersions.v2023_12_01_Preview)
+  @doc("Describes whether profanity was detected.")
+  profanity?: ContentFilterDetectionResult;
+
+  @added(ServiceApiVersions.v2023_12_01_Preview)
+  @projectedName("json", "custom_blocklists")
+  @doc("Describes detection results against configured custom blocklists.")
+  customBlocklists?: ContentFilterBlocklistIdResult[];
+
+  @doc("""
+  Describes an error returned if the content filtering system is
+  down or otherwise unable to complete the operation in time.
+  """)
+  error?: Azure.Core.Foundations.Error;
+};
+
+@added(ServiceApiVersions.v2023_06_01_Preview)
+@doc("Information about content filtering evaluated against input data to Azure OpenAI.")
+model ContentFilterResultDetailsForPrompt {
+  ...ContentFilterResultsBase;
+
+  @doc("Whether a jailbreak attempt was detected in the prompt.")
+  @added(ServiceApiVersions.v2023_12_01_Preview)
+  jailbreak: ContentFilterDetectionResult;
+}
+
+@added(ServiceApiVersions.v2023_06_01_Preview)
+@doc("Information about content filtering evaluated against generated model output.")
+model ContentFilterResultsForChoice {
+  ...ContentFilterResultsBase;
+}
+
+@added(ServiceApiVersions.v2023_06_01_Preview)
+@doc("""
+Content filtering results for a single prompt in the request.
+""")
+model ContentFilterResultsForPrompt {
+  @doc("The index of this prompt in the set of prompt results")
+  @projectedName("json", "prompt_index")
+  promptIndex: int32;
+
+  @doc("Content filtering results for this prompt")
+  @projectedName("json", "content_filter_results")
+  contentFilterResults?: ContentFilterResultDetailsForPrompt;
+}
+
+@added(ServiceApiVersions.v2023_12_01_Preview)
+@doc("Represents the outcome of a detection operation performed by content filtering.")
+model ContentFilterDetectionResult {
+  @doc("A value indicating whether or not the content has been filtered.")
+  filtered: boolean;
+
+  @doc("A value indicating whether detection occurred, irrespective of severity or whether the content was filtered.")
+  detected: string;
```

Comment:
I think this should be a boolean. At least, that's what I'm seeing returned from the API:

```json
{
    "error": {
        "message": "The response was filtered due to the prompt triggering Azure OpenAI's content management policy. Please modify your prompt and retry. To learn more about our content filtering policies please read our documentation: https://go.microsoft.com/fwlink/?linkid=2198766",
        "type": null,
        "param": "prompt",
        "code": "content_filter",
        "status": 400,
        "innererror": {
            "code": "ResponsibleAIPolicyViolation",
            "content_filter_result": {
                "custom_blocklists": [
                    {
                        "filtered": true,
                        "id": "CustomBlockList8"
                    }
                ],
                "hate": {
                    "filtered": false,
                    "severity": "safe"
                },
                "jailbreak": {
                    "detected": false,
                    "filtered": false
                },
                "profanity": {
                    "detected": false,
                    "filtered": false
                },
                "self_harm": {
                    "filtered": false,
                    "severity": "safe"
                },
                "sexual": {
                    "filtered": false,
                    "severity": "safe"
                },
                "violence": {
                    "filtered": false,
                    "severity": "safe"
                }
            }
        }
    }
}
```

Reply:
Thanks for confirming! I suspected that, too--I was trying to faithfully copy from the One True Swagger, but that got a "hmm" as I read it. I'll fix it in both places. @bexxx FYI that the OpenAPI docs have the type of `detected` misrepresented.